c#/Traverse/TraverseFolders/FileSystemVisitor.cs

```diff
@@ -1,72 +1,82 @@
-public delegate bool ReturnedDelegate(string filename, string extention);
+public delegate bool FilterFunctionDelegate(string name);
+
+public delegate void Notify();
+
+public delegate bool MyCustomEvent(string filename,CustomEventArgs eventArgs);
 
 class FileSystemVisitor
 {
+    private readonly string _path;
+    private readonly FilterFunctionDelegate _filterFunc;
+    private readonly CustomEventArgs _eventArgs;
```

Comment:
actually we don't need this "_eventArgs" field here, would be better to pass new instance to event handler every time